File: Algorithm/Searching/Binary Search.cpp
#include<iostream>
using namespace std;
int main(){
  int size; cin>>size;
  int item[size];
  for(int& x:item) cin>>x;
  
  int value; cin>>value;
  int left=0,right=size-1;
  while(left<right){
    int mid=(left+right)/2;
    if(item[mid]==value){
      cout<<"Found"<<endl;
      return 0;
    }
    if(item[mid]>value){
      right=mid;
    }
    else{
      left=mid;
    }
  }
  cout<<"Not Found"<<endl;
  return 0;
}

File: Algorithm/Searching/Binary Search.py
size=int(input(""))
item=list(map(int,input("").split()))
value=int(input(""))
left=0
right=size-1
check=0
while left<=right:
  mid=(left+right)//2
  if item[mid]==value:
    check=1
    break
  if item[mid]>value:
    right=mid-1
  elif item[mid]<value:
    left=mid+1
print("Found" if check else "Not Found")